iina/Preference.swift

```diff
@@ -755,16 +763,16 @@ struct Preference {
     .subOverrideLevel: SubOverrideLevel.strip.rawValue,
     .subTextFont: "sans-serif",
     .subTextSize: Float(55),
-    .subTextColor: NSArchiver.archivedData(withRootObject: NSColor.white),
-    .subBgColor: NSArchiver.archivedData(withRootObject: NSColor.clear),
+    .subTextColorString: convertFromLegacyColor(.subTextColorString) ?? NSColor.white.usingColorSpace(.deviceRGB)!.mpvColorString,
```

Comment:
When I initially read this code it confused me. This is the `defaultPreference` property. It is supposed to contain the default value for this setting. Why would that be anything other than white in `mpv` color string form? This property is used by `AppDelegate` in a call to [register(defaults:)](https://developer.apple.com/documentation/foundation/userdefaults/1417065-register). This is establishing the _default_ values for the setting. But `convertFromLegacyColor` may return a value extracted from the original setting. Confusing. I don't think this causes a problem, but it does not match up with how [UserDefaults](https://developer.apple.com/documentation/foundation/userdefaults) should be used.

Another concern I have about this change has to do with layering and dependencies. `Preference` is one of the lowest level classes. It is even lower than `Logger` as that class must check IINA's settings to see if advanced preferences and logging are enabled. Previously there was no logging in  `Preference`. These changes add logging so now they depend upon each other. Because logging is used while forming the value for `defaultPreference` it means `Logger` is calling into `Preference` before default values have been established as can be seen in the debugger:

<details>
<summary>Stack Trace:</summary>

```text
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 3.1
  * frame #0: 0x000000010246edcc IINA`static Preference.bool(key=(rawValue = "enableAdvancedSettings"), self=IINA.Preference) at Preference.swift:879:12
    frame #1: 0x00000001026058f4 IINA`one-time initialization function for enabled at Logger.swift:56:35
    frame #2: 0x000000010653699c libdispatch.dylib`_dispatch_client_callout + 20
    frame #3: 0x00000001065388bc libdispatch.dylib`_dispatch_once_callout + 160
    frame #4: 0x00000001026059c8 IINA`Logger.enabled.unsafeMutableAddressor at Logger.swift:56:14
    frame #5: 0x00000001026085c4 IINA`static Logger.log(message="Converted color value from legacyKey subTextColor and stored in key subTextColorString", level=debug, subsystem=(rawValue = "iina"), appendNewlineAtTheEnd=true, self=IINA.Logger) at Logger.swift:154:11
    frame #6: 0x000000010246e08c IINA`static Preference.convertFromLegacyColor(key=(rawValue = "subTextColorString"), self=IINA.Preference) at Preference.swift:960:12
    frame #7: 0x000000010246be3c IINA`one-time initialization function for defaultPreference at Preference.swift:766:26
    frame #8: 0x000000010653699c libdispatch.dylib`_dispatch_client_callout + 20
    frame #9: 0x00000001065388bc libdispatch.dylib`_dispatch_once_callout + 160
    frame #10: 0x000000010246e334 IINA`Preference.defaultPreference.unsafeMutableAddressor at Preference.swift:679:14
    frame #11: 0x0000000102650fb8 IINA`AppDelegate.registerUserDefaultValues(self=0x00006000010f04d0) at AppDelegate.swift:755:93
    frame #12: 0x00000001026441e8 IINA`AppDelegate.applicationWillFinishLaunching(notification=Foundation.Notification @ 0x000000016d9d2220, self=0x00006000010f04d0) at AppDelegate.swift:136:5
    frame #13: 0x00000001026464e0 IINA`@objc AppDelegate.applicationWillFinishLaunching(_:) at <compiler-generated>:0
    frame #14: 0x00000001968f30f0 CoreFoundation`__CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 148
    frame #15: 0x000000019698ee7c CoreFoundation`___CFXRegistrationPost_block_invoke + 88
    frame #16: 0x000000019698edc4 CoreFoundation`_CFXRegistrationPost + 440
    frame #17: 0x00000001968c443c CoreFoundation`_CFXNotificationPost + 704
    frame #18: 0x00000001978228a4 Foundation`-[NSNotificationCenter postNotificationName:object:userInfo:] + 88
    frame #19: 0x0000000199b171bc AppKit`-[NSApplication finishLaunching] + 248
    frame #20: 0x0000000199b16ea8 AppKit`-[NSApplication run] + 252
    frame #21: 0x0000000199aee3cc AppKit`NSApplicationMain + 880
    frame #22: 0x00000001026540a0 IINA`main at AppDelegate.swift:24:7
    frame #23: 0x00000001964c7f28 dyld`start + 2236
(lldb)
```
</details>

Another side effect of this is that these new log messages are emitted before `AppDelegate` has had a chance to emit the log file header messages:
```text
14:06:34.834 [iina][d] Converted color value from legacyKey subTextColor and stored in key subTextColorString
14:06:34.836 [iina][d] Converted color value from legacyKey subBgColor and stored in key subBgColorString
14:06:34.837 [iina][d] Converted color value from legacyKey subBorderColor and stored in key subBorderColorString
14:06:34.840 [iina][d] IINA 1.3.1 Build 133
14:06:34.840 [iina][d] Copyright © 2017-2022
14:06:34.840 [iina][d] Collider LI, et al.
14:06:34.840 [iina][d] Released under GPLv3.
```

I'm thinking the calls to `convertFromLegacyColor` should be removed from the initialization of `defaultPreference` and into a `migrateSettings` function that `AppDelegate` would then call in `applicationWillFinishLaunching` right before emitting the "App will launch" log message.